addons/quicktime/fnc_runQTE.sqf

```diff
@@ -0,0 +1,94 @@
+#include "script_component.hpp"
+/* ----------------------------------------------------------------------------
+Function: CBA_fnc_runQTE
+
+Description:
+    Runs a Quick-Time Event.
+
+Parameters:
+    _object - <OBJECT>
+    _args - Extra arguments passed to the _on... functions<ARRAY>
+    _onDisplay - Code callback on displayable event passed [_args, _qteSequence, _qteHistory]. <CODE, STRING>
+    _onFinish - Code callback on Quick-Time Event completed passed [_args, _elapsedTime]. <CODE, STRING>
+    _onFinish - Code callback on Quick-Time Event timeout/outranged passed [_args, _elapsedTime]. <CODE, STRING>
+    _qteSequence - Quick-Time sequence made up of ["↑", "↓", "→", "←"] <ARRAY>
+    _maxDistance - max interaction distance from attached object <NUMBER> (default: 10) 
+    _timeout - ingame timeout <NUMBER> (default: 30)
+
+Example:
+    [car,
+    [], 
+    { 
+        hint format [
+            "%1 \n %2",
+            [_this select 1] call CBA_fnc_getFormattedQTESequence,
+            [_this select 2] call CBA_fnc_getFormattedQTESequence
+        ]
+    }, 
+    { 
+        hint "Finished!"; 
+    },
+    { 
+        hint "Failure!"; 
+    },
+    ["↑", "↓", "→", "←"]] call CBA_fnc_runQTE
+
+Returns:
+    True if the QTE was started, false if it was already running <BOOELAN>
+
+Author:
+    john681611
+---------------------------------------------------------------------------- */
+
+
+params ["_object", "_args", "_onDisplay", "_onFinish", "_onFail", "_qteSequence", ["_maxDistance", 10], ["_timeout", 30]];
```

Comment:
Since this is a public function, we need to sanitise the input:
```suggestion
params [["_object", objNull, [objNull]], "_args", ["_onDisplay", {}, ["", {}]], ["_onFinish", {}, ["", {}]], ["_onFail", {}, ["", {}]], ["_qteSequence", [], [[]]], ["_maxDistance", 10, [0]], ["_timeout", 30, [0]]];
```

You need to check if the input are valid (e.g. an empty QTE sequence should not be allowed).

I haven't taken into account the other proposals I've made, so you'll need to adapt accordingly.